BraintreeCore/src/main/java/com/braintreepayments/api/AnalyticsClient.kt

```diff
@@ -0,0 +1,220 @@
+        }
+    }
+
+    private fun scheduleAnalyticsUpload(
+        configuration: Configuration,
+        authorization: Authorization,
+        sessionId: String?,
+        integration: String?
+    ): UUID {
+        val inputData = workDataOf(
+            WORK_INPUT_KEY_AUTHORIZATION to authorization.toString(),
+            WORK_INPUT_KEY_CONFIGURATION to configuration.toJson(),
+            WORK_INPUT_KEY_SESSION_ID to sessionId,
+            WORK_INPUT_KEY_INTEGRATION to integration
+        )
+        val analyticsWorkRequest = OneTimeWorkRequest.Builder(AnalyticsUploadWorker::class.java)
+            .setInitialDelay(30, TimeUnit.SECONDS)
+            .setInputData(inputData)
+            .build()
+        workManager.enqueueUniqueWork(
+            WORK_NAME_ANALYTICS_UPLOAD, ExistingWorkPolicy.KEEP, analyticsWorkRequest
+        )
+        return analyticsWorkRequest.id
+    }
+
+    fun uploadAnalytics(context: Context?, inputData: Data): ListenableWorker.Result {
+        val configuration = getConfigurationFromData(inputData)
+        val authorization = getAuthorizationFromData(inputData)
+        val sessionId = inputData.getString(WORK_INPUT_KEY_SESSION_ID)
+        val integration = inputData.getString(WORK_INPUT_KEY_INTEGRATION)
+        val isMissingInputData =
+            listOf(configuration, authorization, sessionId, integration).contains(null)
+        return if (isMissingInputData) {
+            ListenableWorker.Result.failure()
+        } else try {
+            val analyticsEventDao = analyticsDatabase.analyticsEventDao()
+            val events = analyticsEventDao.getAllEvents()
+            if (events.isNotEmpty()) {
+                val metadata = deviceInspector.getDeviceMetadata(context, sessionId, integration)
+                val analyticsRequest = serializeEvents(authorization, events, metadata)
+                val analyticsUrl = configuration!!.analyticsUrl
+                httpClient.post(
+                    analyticsUrl, analyticsRequest.toString(), configuration, authorization
+                )
+                analyticsEventDao.deleteEvents(events)
+            }
+            ListenableWorker.Result.success()
+        } catch (e: Exception) {
+            ListenableWorker.Result.failure()
+        }
+    }
+
+    fun reportCrash(
+        context: Context?, sessionId: String?, integration: String?, authorization: Authorization?
+    ) {
+        reportCrash(context, sessionId, integration, System.currentTimeMillis(), authorization)
+    }
+
+    @VisibleForTesting
+    fun reportCrash(
+        context: Context?,
+        sessionId: String?,
+        integration: String?,
+        timestamp: Long,
+        authorization: Authorization?
+    ) {
+        if (lastKnownAnalyticsUrl == null || authorization == null) {
+            return
+        }
+        val metadata = deviceInspector.getDeviceMetadata(context, sessionId, integration)
+        val event = AnalyticsEvent("android.crash", timestamp)
```

Comment:
Interesting - I didn't realize we were doing this on Android. We don't have an equivalent on iOS. Is this something we've ever used to troubleshoot merchant issues?

Reply:
I have a feeling it's for tracking. I've never seen it come up in an inbound. It was weird to write when we ported `v3` to `v4` because it relies on the presence of an analytics url, which means it'll only work if it has fetched configuration before. Another guess here, it's probably for tracking purposes so we have some insight into app crashes. But honestly these may be available to us in the Google Play SDK console now. It would be cool to remove it in the next major version if possible.

Reply:
Yeh, I've also never seen it come up or be used. Definitely worth consider removing in V5